### PR TITLE
Add localization from OneLocBuild (2021-05-07)

### PR DESCRIPTION
**Pull request by jonfortescue**
Among other things, this fixes [AB#1321419](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1321419)

### CODE REVIEW

File: src/Analyzers/CSharp/Analyzers/xlf/CSharpAnalyzersResources.de.xlf [hunk 1]
```diff
@@ -14,12 +14,12 @@
...
       <trans-unit id="Blank_line_not_allowed_after_constructor_initializer_colon">
         <source>Blank line not allowed after constructor initializer colon</source>
-        <target state="new">Blank line not allowed after constructor initializer colon</target>
+        <target state="translated">Nach dem Doppelpunkt für den Initialisierer des Konstruktors ist keine leere Zeile zulässig.</target>
         <note />
       </trans-unit>
       <trans-unit id="Consecutive_braces_must_not_have_a_blank_between_them">
         <source>Consecutive braces must not have blank line between them</source>
-        <target state="new">Consecutive braces must not have blank line between them</target>
+        <target state="translated">Aufeinanderfolgende geschweifte Klammern dürfen keine leere Zeile einschließen.</target>
```
Comment (JoeRobich) [inline src/Analyzers/CSharp/Analyzers/xlf/CSharpAnalyzersResources.de.xlf:22]: Same issue with ending punctuation.

File: src/Analyzers/CSharp/Analyzers/xlf/CSharpAnalyzersResources.cs.xlf [hunk 2]
```diff
@@ -14,12 +14,12 @@
...
       <trans-unit id="Blank_line_not_allowed_after_constructor_initializer_colon">
         <source>Blank line not allowed after constructor initializer colon</source>
-        <target state="new">Blank line not allowed after constructor initializer colon</target>
+        <target state="translated">Za dvojtečkou inicializátoru konstruktoru se nepovoluje prázdný řádek.</target>
         <note />
       </trans-unit>
       <trans-unit id="Consecutive_braces_must_not_have_a_blank_between_them">
         <source>Consecutive braces must not have blank line between them</source>
-        <target state="new">Consecutive braces must not have blank line between them</target>
+        <target state="translated">Po sobě jdoucí závorky nesmí mít mezi sebou prázdný řádek.</target>
```
[the remark this answers is not in the document]
Comment (jonfortescue) [reply src/Analyzers/CSharp/Analyzers/xlf/CSharpAnalyzersResources.cs.xlf:22]: @cristianosuzuki77

[the remark this answers is not in the document]
Comment (JoeRobich) [reply src/Analyzers/CSharp/Analyzers/xlf/CSharpAnalyzersResources.cs.xlf:22]: Not sure if it is helpful for me to point out each occurence of this but it occurs in many other strings in many of the .xlf files being changed.

[the remark this answers is not in the document]
Comment (jonfortescue) [reply src/Analyzers/CSharp/Analyzers/xlf/CSharpAnalyzersResources.cs.xlf:22]: I'd say leave them for now. Cristiano will report them and they'll be fixed in a future update.

[the remark this answers is not in the document]
Comment (cristianosuzuki77) [reply src/Analyzers/CSharp/Analyzers/xlf/CSharpAnalyzersResources.cs.xlf:22]: Unless specified in the note section (by adding comments in the resx), translations will use the appropriated rules of the local grammar. Some languages will require punctuation in complete sentences.
Do you need the punctuation out?

**Comment by wli3** [comment]
@JoeRobich please merge it now. For actual translation error please follow up with @cristianosuzuki77

File: src/Analyzers/CSharp/Analyzers/xlf/CSharpAnalyzersResources.cs.xlf (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoeRobich) [reply src/Analyzers/CSharp/Analyzers/xlf/CSharpAnalyzersResources.cs.xlf:22]: We could add notes for each string whether it is intended to be used on its own or combined into a more complete string. As a rule, we add punctuation when we are intending strings to be complete sentences. In this case, these are fragments that are concatenated together to form diagnostic messages.

Do you recommend we start marking up each string? Is that how other projects are handling this?

[the remark this answers is not in the document]
Comment (JoeRobich) [reply src/Analyzers/CSharp/Analyzers/xlf/CSharpAnalyzersResources.cs.xlf:22]: @cristianosuzuki77

**Pull request merged**